### PR TITLE
Make commands 

### LINKED ISSUE #14: Add regular commands thing

### PR DESCRIPTION
**Pull request by ToxicKidz**
## So...
This closes #14 and yeah that's it. The code is pretty bad but ¯\\_(ツ)_/¯

### CODE REVIEW

File: snakecord/commands/commander.py [hunk 1]
```diff
@@ -6,3 +6,32 @@ def __init__(self, commander, message, parser, prefix):
...
+        except EOFError:
+            return
+        for name, command in commander.commands.items():
+            if cmd == name:
+                self.command = command
+    
+    def invoke(self):
+        if not self.command:
+            return
+        self.command.invoke(self)
```
Comment (ToxicKidz) [inline snakecord/commands/commander.py]:
```suggestion
        if self.command is not None:
            self.command.invoke(self)
```

File: snakecord/commands/commander.py [hunk 2]
```diff
@@ -6,3 +6,32 @@ def __init__(self, commander, message, parser, prefix):
...
+class CommandInvokeHandler:
+    def __init__(self, name):
+        self.name = name
+    
+    def __call__(self, commander, evnt, *args, **kwargs):
+        self.commander = commander
+        self.evnt = evnt
+        self.args = args
+        self.kwargs = kwargs
+        return self
```
Comment (ToxicKidz) [inline snakecord/commands/commander.py:37]:
```suggestion
        return self

```

File: snakecord/commands/client.py [hunk 3]
```diff
@@ -1,8 +1,58 @@
...
 from .parsers import StringParser, FunctionArgParser
+from ..events import EventPusher
+from .commander import CommandInvokeEvent, CommandInvokeHandler
+from .command import Command
 
+from asyncio import get_event_loop
+
+class Commander(EventPusher):
+    def __init__(self, **kwargs):
+        loop = kwargs.get("loop") or get_event_loop()
```
Comment (ToxicKidz) [inline snakecord/commands/client.py]:
```suggestion
    def __init__(self, loop=None):
        loop = loop or get_event_loop()
```
What was I even thinking 🤔

**Review by ToxicKidz — COMMENTED** [review]
Reviewed my own PR because sleep-cult didn't.

**Comment by asleep-cult** [comment]
Hi, sorry for taking so long to reply. I think it's important that we take our time when implementing something like a commands extension because it's a very significant part of the library. This implementation just feels too rushed and I'm not sure what I would want a commands extension to look like at the moment. Now that slash commands are a thing I am left wondering a. Is it still worth it to implement a commands extension and b. How can slash commands and regular commands be implemented together? I also don't think the library is mature enough for us to start working on a commands extension, almost every commit is a breaking change and there are multiple bugs.

TL;DR: This library isn't ready for a commands extension.

**Comment by 15696** [comment]
howdo reopen
